scipy/signal/tests/test_max_len_seq.py

```diff
@@ -39,18 +41,23 @@ def test_mls_output(self):
                     m = 2. * orig_m - 1.  # convert to +/- 1 representation
                     # First, make sure we got all 1's or -1
                     err_msg = "mls had non binary terms"
-                    assert_array_equal(np.abs(m), np.ones_like(m),
+                    xp_assert_equal(np.abs(m), np.ones_like(m),
                                        err_msg=err_msg)
                     # Test via circular cross-correlation, which is just mult.
                     # in the frequency domain with one signal conjugated
                     tester = np.real(ifft(fft(m) * np.conj(fft(m))))
                     out_len = 2**nbits - 1
                     # impulse amplitude == test_len
                     err_msg = "mls impulse has incorrect value"
-                    assert_allclose(tester[0], out_len, err_msg=err_msg)
+                    xp_assert_close(tester[0],
+                                    np.asarray(out_len, dtype=np.float64),
+                                    check_0d=False,
+                                    err_msg=err_msg
+                    )
```

Comment:
is the case here that `tester[0]` returns a scalar, but we need to use `asarray` to cast `out_len` to the correct dtype?

Reply:
yes:

```
(Pdb) p tester[0].__class__
<class 'numpy.float64'>
(Pdb) p out_len.__class__
<class 'int'>
```

Reply:
Thanks. Would we be able to instead write this as:

```suggestion
                    xp_assert_close(tester[0],
                                    float(out_len),
                                    err_msg=err_msg
                    )
```

or does something go wrong there?